### PR TITLE
PR: Enable CircleCI and first tests addition

### LINKED ISSUE #24: Add tests to the project and CI Integration

### LINKED ISSUE #27: Consistent naming with other spyder plugins
May I suggests to change the package name from `spyder.vim` to `spyder-vim`, to be consistent with other published plugins such as `spyder-unittest`?

FYI, the name `spyder.vim` suggests either a library laid out as `spyder/vim`, a namespaced package under the `spyder` umbrella, or a plugin for `vim`, neither of which is the case.

### PR DESCRIPTION
**Pull request by andfoy**
Fixes #24 
Fixes #27

### CODE REVIEW

File: circle.yml [hunk 1]
```diff
@@ -0,0 +1,52 @@
+# https://circleci.com/gh/spyder-ide/spyder-vim/
+
+machine:
+  environment:
+    # Python versions to tests (Maximum of 4 different versions)
+    # The last container is used to test with pyqt5 wheels
+    PY_VERSIONS: "2.7 3.6 3.5"
+    # For Coveralls
+    COVERALLS_REPO_TOKEN: sTGxv5iQxuiBQCoxLB53t6m2xSe58rAIq
```
Comment (rlaverde) [inline circle.yml]: This token should be deleted form here (and regenerated), and added instead to https://circleci.com/gh/spyder-ide/spyder-vim/edit#env-vars

File: .checkignore [hunk 2]
```diff
@@ -0,0 +1,16 @@
+# Quantified code ignore file
```
Comment (ccordoba12) [inline .checkignore]: Please remove this file. Quantified Code was closed.

Comment (rlaverde) [inline .checkignore]: Maybe we could use Scrutinizer as qtpy does (It's somehow similar)

**Comment by ccordoba12** [comment]
Now you need to fix all style issue reported by `ciocheck` ;-)

File: setup.py [hunk 3]
```diff
@@ -1,38 +1,67 @@
...
+        data = f.read()
+    return data
+
+
+REQUIREMENTS = ['spyder>=3.2.0']
 
 setup(
-    name='spyder.vim',
-    version="0.1.0",
+    name='spyder_vim',
```
Comment (ccordoba12) [inline setup.py]: This needs to be `spyder-vim`, not `spyder_vim`

File: setup.py [hunk 4]
```diff
@@ -1,38 +1,67 @@
...
+    version=get_version(),
     packages=find_packages(exclude=['contrib', 'docs', 'tests*']),
-    keywords=["spyder ide plugin addon vi vim"],
-    url='https://github.com/Nodd/spyderplugins.vim',
+    keywords=['Spyder', 'Plugin'],
```
Comment (ccordoba12) [inline setup.py]: `Vim` is missing here.

File: setup.py (hunk 3 above)
Comment (andfoy) [inline setup.py]: Then I should also update spyder-terminal then!

Comment (ccordoba12) [inline setup.py]: Please open an issue about it before doing it. I don't know if it's possible at this point.

Comment (andfoy) [inline setup.py]: It seems that the package is right: https://pypi.python.org/pypi/spyder-terminal

Comment (ccordoba12) [inline setup.py]: Great! Then it's probably safe to do it ;-)

**Pull request merged**